pkg/server/events.go

```diff
@@ -17,72 +17,69 @@ limitations under the License.
 package server
 
 import (
-	"time"
+	"fmt"
 
 	"github.com/containerd/containerd"
 	"github.com/containerd/containerd/api/services/events/v1"
 	"github.com/containerd/containerd/errdefs"
 	"github.com/containerd/containerd/typeurl"
 	"github.com/golang/glog"
-	"github.com/jpillora/backoff"
 	"golang.org/x/net/context"
 
 	containerstore "github.com/kubernetes-incubator/cri-containerd/pkg/store/container"
 )
 
-const (
-	// minRetryInterval is the minimum retry interval when lost connection with containerd.
-	minRetryInterval = 100 * time.Millisecond
-	// maxRetryInterval is the maximum retry interval when lost connection with containerd.
-	maxRetryInterval = 30 * time.Second
-	// exponentialFactor is the exponential backoff factor.
-	exponentialFactor = 2.0
-)
-
-// startEventMonitor starts an event monitor which monitors and handles all
-// container events.
+// eventMonitor monitors containerd event and updates internal state correspondingly.
 // TODO(random-liu): [P1] Is it possible to drop event during containerd is running?
-func (c *criContainerdService) startEventMonitor() {
-	b := backoff.Backoff{
-		Min:    minRetryInterval,
-		Max:    maxRetryInterval,
-		Factor: exponentialFactor,
+type eventMonitor struct {
+	c           *criContainerdService
+	eventstream events.Events_SubscribeClient
+	cancel      context.CancelFunc
+	close       chan struct{}
```

Comment:
nit: can we rename to close to closeChan or something on those lines.

Reply:
Will do.